### PR TITLE
Add tests for taproot descriptors having deep scripts trees.

### PR DESCRIPTION
**Pull request by kanishk779**
- As we know that there is a limit to the depth of the script tree in taproot specification.
- These tests try to create script trees with depths 127 and 128. The limit is 128 (i.e a script tree of depth 129 is invalid) for valid trees.
- Another test with a complete tree having 128 leaves has also been added.
- A simple python script was used to generate these test cases, [here](https://gist.github.com/kanishk779/a4829eab5e60d35215de5e2665b18b7a) it is.

### CODE REVIEW

File: integration_test/src/test_desc.rs [hunk 1]
```diff
@@ -104,8 +104,10 @@ pub fn test_desc_satisfy(cl: &Client, testdata: &TestData, desc: &str) -> Witnes
     let addr = cl
         .get_new_address(None, Some(json::AddressType::Bech32))
         .unwrap();
+    // Had to decrease 'value', so that fees can be increased 
+    // (Was getting insufficient fees error, for deep script trees)
     psbt.unsigned_tx.output.push(TxOut {
-        value: 99_999_000,
+        value: 99_997_000,
```
Comment (sanket1729) [inline integration_test/src/test_desc.rs:110]: Interesting, thanks for debugging. Will open this up with a good first issue.

File: integration_test/src/main.rs [hunk 2]
```diff
@@ -111,6 +111,16 @@ fn test_descs(cl: &Client, testdata: &TestData) {
...
...pk(X27!),{pk(X28!),{pk(X29!),{pk(X30!),{pk(X31!),{pk(X32!),{pk(X33!),{pk(X34!),{pk(X35!),{pk(X36!),{pk(X37!),{pk(X38!),{pk(X39!),{pk(X40!),{pk(X41!),{pk(X42!),{pk(X43!),{pk(X44!),{pk(X45!),{pk(X46!),{pk(X47!),{pk(X48!),{pk(X49!),{pk(X50!),{pk(X51!),{pk(X52!),{pk(X53!),{pk(X54!),{pk(X55!),{pk(X56!),{pk(X57!),{pk(X58!),{pk(X59!),{pk(X60!),{pk(X61!),{pk(X62!),{pk(X63!),{pk(X64!),{pk(X65!),{pk(X66!),{pk(X67!),{pk(X68!),{pk(X69!),{pk(X70!),{pk(X71!),{pk(X72!),{pk(X73!),{pk(X74!),{pk(X75!),{pk(X76!),{pk(X77!),{pk(X78!),{pk(X79!),{pk(X80!),{pk(X81!),{pk(X82!),{pk(X83!),{pk(X84!),{pk(X85!),{pk(X86!),{pk(X87!),{pk(X88!),{pk(X89!),{pk(X90!),{pk(X91!),{pk(X92!),{pk(X93!),{pk(X94!),{pk(X95!),{pk(X96!),{pk(X97!),{pk(X98!),{pk(X99!),{pk(X100!),{pk(X101!),{pk(X102!),{pk(X103!),{pk(X104!),{pk(X105!),{pk(X106!),{pk(X107!),{pk(X108!),{pk(X109!),{pk(X110!),{pk(X111!),{pk(X112!),{pk(X113!),{pk(X114!),{pk(X115!),{pk(X116!),{pk(X117!),{pk(X118!),{pk(X119!),{pk(X120!),{pk(X121!),{pk(X122!),{pk(X123!),{pk(X124!),{pk(X125!),{pk(X126!),{pk(X127!),{pk(X128!),{pk(X129!),pk(X130)}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}})");
```
Comment (sanket1729) [inline integration_test/src/main.rs:119]: Did you also add a commented test here? This does not belong in this commit.

**Review by sanket1729 — COMMENTED** [review]
One small comment

File: integration_test/src/main.rs (hunk 2 above)
Comment (kanishk779) [reply integration_test/src/main.rs:119]: Yes, I should have removed it. This was for testing with depth 129 trees to check whether they are invalid or not.

**Comment by sanket1729** [comment]
Can you make sure there is only commit here? @kanishk779. It might require you to read online about `git squash` and `fixup`.

**Review by apoelstra — APPROVED** [review]
ACK 1ff13d408f15e130fa192ce9cbd1f20f26c54b6f

Thanks! These are cool tests.

File: integration_test/src/main.rs [hunk 3]
```diff
@@ -111,6 +111,15 @@ fn test_descs(cl: &Client, testdata: &TestData) {
...
+
```
Comment (sanket1729) [inline integration_test/src/main.rs:119]: Unneeded Whitespace here

File: integration_test/src/test_desc.rs [hunk 4]
```diff
@@ -104,8 +104,10 @@ pub fn test_desc_satisfy(cl: &Client, testdata: &TestData, desc: &str) -> Witnes
     let addr = cl
         .get_new_address(None, Some(json::AddressType::Bech32))
         .unwrap();
+    // Had to decrease 'value', so that fees can be increased
```
Comment (sanket1729) [inline integration_test/src/test_desc.rs:107]: Whiespace here. Will merge this now, but please take of this in future. Or address in a later PR.

**Review by sanket1729 — APPROVED** [review]
ACK 1ff13d408f15e130fa192ce9cbd1f20f26c54b6f
